File: C++/11.ContainerWithMostWater.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxArea = 0;
        int l = 0, r = height.size() - 1;
        while (l < r) {
            int tempHeight = min(height[l], height[r]) * (r - l);
            maxArea = max(tempHeight, maxArea);
            if (height[r] > height[l]) {
                l++;
            } else {
                r--;
            }
        }
        return maxArea;
    }
};

File: Swift/11ContainerWithMostWater.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var l = 0, r = height.count - 1, maxArea = 0

        while l < r {
            let tempArea = min(height[l], height[r]) * (r - l)
            maxArea = max(maxArea, tempArea)
            if height[r] > height[l] {
                l += 1
            } else {
                r -= 1
            }
        }
        return maxArea
    }
}
